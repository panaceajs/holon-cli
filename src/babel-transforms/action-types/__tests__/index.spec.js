const pluginTester = require('babel-plugin-tester');
const actionTypes = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: actionTypes,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should add nothing when there is nothing to add.',
      fixture: '__fixtures__/without-exports.js',
      snapshot: false
    },
    {
      title: 'it should add action type exports.',
      fixture: '__fixtures__/without-exports.js',
      pluginOptions: { actionTypes: { someAction: 'someAction' } }
    },
    {
      title: 'it should update action type exports.',
      fixture: '__fixtures__/with-exports.js',
      pluginOptions: {
        actionTypes: {
          existingAction: 'existingAction',
          nonExistingAction: 'nonExistingAction'
        }
      }
    }
  ]
});
