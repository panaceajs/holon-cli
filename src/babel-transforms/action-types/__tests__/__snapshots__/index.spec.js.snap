// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-action-types it should add action type exports.: it should add action type exports. 1`] = `
"
console.log('nothing going on here');

      ↓ ↓ ↓ ↓ ↓ ↓

export const SOME_ACTION = 'SOME_ACTION';
console.log('nothing going on here');
"
`;

exports[`holon-action-types it should update action type exports.: it should update action type exports. 1`] = `
"
export const EXISTING_ACTION = 'EXISTING_ACTION';
console.log('nothing going on here');

      ↓ ↓ ↓ ↓ ↓ ↓

export const EXISTING_ACTION = 'EXISTING_ACTION';
export const NON_EXISTING_ACTION = 'NON_EXISTING_ACTION';
console.log('nothing going on here');
"
`;
